sixtyfps_runtime/corelib/graphics/image.rs

```diff
@@ -54,39 +270,80 @@ impl Image {
     pub fn load_from_path(path: &std::path::Path) -> Result<Self, LoadImageError> {
         Ok(Image(ImageInner::AbsoluteFilePath(path.to_str().ok_or(LoadImageError(()))?.into())))
     }
-    /*
-    /// Create an image from argb pixels.
+
+    /// Creates a new Image from the specified shared pixel buffer, where each pixel has three color
+    /// channels (red, green and blue) encoded as u8.
+    pub fn from_rgb8(buffer: SharedPixelBuffer<RGB8Pixel>) -> Self {
+        Image(ImageInner::EmbeddedImage(SharedImageBuffer::RGB8(buffer)))
+    }
+
+    /// Creates a new Image from the specified shared pixel buffer, where each pixel has four color
+    /// channels (red, green, blue and alpha) encoded as u8.
+    pub fn from_rgba8(buffer: SharedPixelBuffer<RGBA8Pixel>) -> Self {
+        Image(ImageInner::EmbeddedImage(SharedImageBuffer::RGBA8(buffer)))
+    }
+
+    /// Creates a new Image from the specified shared pixel buffer, where each pixel has four color
+    /// channels (red, green, blue and alpha) encoded as u8 and, in contrast to [`Self::from_rgba8`],
+    /// the alpha channel is also assumed to be multiplied to the red, green and blue channels.
     ///
-    /// Returns an error if the `data` does not have a size of `width * height`.
-    pub fn load_from_argb(
-        width: u32,
-        height: u32,
-        data: SharedVector<u32>,
-    ) -> Result<Self, LoadImageError> {
-        if data.len() != width as usize * height as usize {
-            Err(LoadImageError(()))
-        } else {
-            Ok(Image(ImageReference::EmbeddedRgbaImage { width, height, data }))
-        }
+    /// Only construct an Image with this function if you know that your pixels are encoded this way.
```

Comment:
What if it isn't?  Is it UB? Then the function should be unsafe. Or is it just that were gonna get rendering artifacts? Or should we check it and panic if it isn't?

Reply:
There is no way of programmatically checking if the alpha is pre-multiplied or not. If you use this function with an RGBA image that has the alpha not pre-multiplied, then you'll get a visually differently rendered image (it'll have a kind of tint) - and vice versa: If you get an Rgba-premultiplied image and use it with `from_rgba8` you'll see that the colors are off.